src/stackdriver-nozzle/heartbeat/heartbeater.go

```diff
@@ -0,0 +1,73 @@
+package heartbeat
+
+import (
+	"errors"
+	"time"
+
+	"github.com/cloudfoundry/lager"
+)
+
+type Heartbeater interface {
+	Start()
+	Increment(string)
+	Stop()
+}
+
+type heartbeater struct {
+	logger  lager.Logger
+	trigger <-chan time.Time
+	counter chan string
+	done    chan struct{}
+	started bool
+}
+
+func NewHeartbeater(logger lager.Logger, trigger <-chan time.Time) Heartbeater {
```

Comment:
nit: New would be more idiomatic but I know we use this style in other places so feel free to ignore